meson.build

```diff
@@ -0,0 +1,248 @@
+conf.set_quoted('BINDIR',		bindir)
+conf.set_quoted('LIBDIR',		libdir)
+conf.set_quoted('LOCALSTATEDIR',	localstatedir)
+conf.set_quoted('LXCFSSHAREDIR',	join_paths(localstatedir, 'lxcfs'))
+conf.set_quoted('LXCCONFDIR',		join_paths(localstatedir, '/lxc/config/common.conf.d'))
+conf.set_quoted('RUNTIME_PATH',		runtimepath)
+conf.set_quoted('SYSCONFDIR',		sysconfdir)
+
+init_script = get_option('init-script')
+
+cc = meson.get_compiler('c')
+meson_build_sh = find_program('tools/meson-build.sh')
+
+want_tests = get_option('tests')
+
+possible_cc_flags = [
+	'-fPIE',
+	'-Wvla',
+	'-std=gnu11',
+	'-fdiagnostics-color',
+	'-Wimplicit-fallthrough=5',
+	'-Wcast-align',
+	'-Wstrict-prototypes',
+	'-fno-strict-aliasing',
+	'-fstack-clash-protection',
+	'-fstack-protector-strong',
+	'--param=ssp-buffer-size=4',
+	'--mcet -fcf-protection',
+	'-Werror=implicit-function-declaration',
+	'-Wlogical-op',
+	'-Wmissing-include-dirs',
+	'-Wold-style-definition',
+	'-Winit-self',
+	'-Wunused-but-set-variable',
+	'-Wno-unused-parameter',
+	'-Wfloat-equal',
+	'-Wsuggest-attribute=noreturn',
+	'-Werror=return-type',
+	'-Werror=incompatible-pointer-types',
+	'-Wformat=2',
+	'-Wshadow',
+	'-Wendif-labels',
+	'-Werror=overflow',
+	'-fdiagnostics-show-option',
+	'-Werror=shift-count-overflow',
+	'-Werror=shift-overflow=2',
+	'-Wdate-time',
+	'-Wnested-externs',
+	'-fasynchronous-unwind-tables',
+	'-fexceptions',
+	'-Warray-bounds',
+	'-Wrestrict',
+	'-Wreturn-local-addr',
+	'-fsanitize=cfi',
+	'-Wstringop-overflow',
+]
+
+possible_link_flags = [
+	'-Wl,--as-needed',
+	'-Wl,--gc-sections',
+	'-Wl,-z,relro',
+	'-Wl,-z,now',
+	'-pie',
+	'-Wl,-fuse-ld=gold',
+]
+
+if meson.version().version_compare('>=0.46')
+	add_project_link_arguments(cc.get_supported_link_arguments(possible_link_flags), language : 'c')
+else
+	add_project_link_arguments(possible_link_flags, language : 'c')
+endif
+
+add_project_arguments(cc.get_supported_arguments(possible_cc_flags), language : 'c')
+
+foreach ident : [
+        ['strlcpy',		'''#include <string.h>'''],
+        ['strlcat',		'''#include <string.h>'''],
+        ['pidfd_send_signal',	'''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['pidfd_open',		'''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+]
+
+        have = cc.has_function(ident[0], prefix : ident[1], args : '-D_GNU_SOURCE')
+        conf.set10('HAVE_' + ident[0].to_upper(), have)
+endforeach
+
+conf.set('_FILE_OFFSET_BITS', 64)
+
+libfuse = cc.find_library('fuse3', required : false)
+if libfuse.found()
+	conf.set10('HAVE_FUSE3', libfuse.found())
+	conf.set('FUSE_USE_VERSION', 30)
+else
+	libfuse = cc.find_library('fuse')
+	conf.set10('HAVE_FUSE', libfuse.found())
```

Comment:
It doesn't seem this define is ever used, it is just detected by virtue of the "else" clause in variious HAVE_FUSE3 checks.

Reply:
I reworked this, thanks!